--- autogen-5.18.3/autoopts/mk-tpl-config.sh.orig	2014-05-28 20:19:54.591978186 +0200
+++ autogen-5.18.3/autoopts/mk-tpl-config.sh	2014-05-28 20:03:53.698664880 +0200
@@ -200,9 +200,11 @@
     cd ${builddir}
     find_libguiledir "${LGCFLAGS}"
 
+    set +e
     list=`set +e ; exec 2>/dev/null
         find ${libguiledir}/libguile* -type f | \
             xargs grep -l -E '\<noreturn\>'`
+    set -e
 
     test -z "$list" && exit 0
 
