--- autogen-5.18/doc/auto-opts.tlib.orig	2013-07-21 16:30:24.730920497 +0200
+++ autogen-5.18/doc/auto-opts.tlib	2013-07-21 16:37:12.804236705 +0200
@@ -106,8 +106,8 @@
 \=]
 base=check
 BASE=`echo $base | tr '[a-z-]' '[A-Z_]'`
-cflags="-DTEST_${BASE} `autoopts-config cflags`"
-ldflags="`autoopts-config ldflags`"
+cflags="-DTEST_${BASE} "
+ldflags="-L${top_builddir}/autoopts/.libs -lopts"
 autogen ${base}.def
 cc -o ${base} -g ${cflags} ${base}.c ${ldflags}
 ./${base} --help
